Spigot-Server-Patches/0600-Allow-toggling-special-MobSpawners-per-world.patch

```diff
@@ -0,0 +1,75 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: jmp <jasonpenilla2@me.com>
+Date: Wed, 26 Aug 2020 11:00:52 -0700
+Subject: [PATCH] Allow toggling special MobSpawners per world
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index cb19a38f2082ed7a46580c1e8989113a3eae9135..e0f68a8bde1787930b4367f195495d4a16a324e2 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -196,6 +196,21 @@ public class PaperWorldConfig {
+         mobSpawnerTickRate = getInt("mob-spawner-tick-rate", 1);
+     }
+ 
++    public boolean catSpawning;
++    public boolean patrolSpawning;
++    public boolean phantomSpawning;
++    public boolean villagerTraderSpawning;
++    public boolean villageSiegeSpawning;
++    private void mobSpawnerSettings() {
++        // values of "default" or null will default to true only if the world environment is normal (aka overworld)
++        Predicate<Boolean> predicate = (bool) -> (bool != null && bool) || (bool == null && environment == Environment.NORMAL);
++        catSpawning = getBoolean("mob-spawning.village-cats", predicate);
++        patrolSpawning = getBoolean("mob-spawning.raid-patrols", predicate);
++        phantomSpawning = getBoolean("mob-spawning.phantoms", predicate);
++        villagerTraderSpawning = getBoolean("mob-spawning.wandering-traders", predicate);
++        villageSiegeSpawning = getBoolean("mob-spawning.village-sieges", predicate);
++    }
++
+     public int containerUpdateTickRate;
+     private void containerUpdateTickRate() {
+         containerUpdateTickRate = getInt("container-update-tick-rate", 1);
+diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
+index d0ffe5790e98b65cd6035ad529e7da8172de9029..9f90cf4c9da8b870eb6cbc468356d5a82035b2c6 100644
+--- a/src/main/java/net/minecraft/server/MinecraftServer.java
++++ b/src/main/java/net/minecraft/server/MinecraftServer.java
+@@ -402,7 +402,7 @@ public abstract class MinecraftServer extends IAsyncTaskHandlerReentrant<TickTas
+             boolean flag = generatorsettings.isDebugWorld();
+             long i = generatorsettings.getSeed();
+             long j = BiomeManager.a(i);
+-            List<MobSpawner> list = ImmutableList.of(new MobSpawnerPhantom(), new MobSpawnerPatrol(), new MobSpawnerCat(), new VillageSiege(), new MobSpawnerTrader(iworlddataserver));
++            List<MobSpawner> list = ImmutableList.of(new MobSpawnerPhantom(), new MobSpawnerPatrol(), new MobSpawnerCat(), new VillageSiege(), new MobSpawnerTrader(iworlddataserver)); // Paper - This list is unused as we make these configurable in `Allow toggling special MobSpawners per world`. If this list changes, that patch likely needs updating.
```

Comment:
```suggestion
+            // List<MobSpawner> list = ImmutableList.of(new MobSpawnerPhantom(), new MobSpawnerPatrol(), new MobSpawnerCat(), new VillageSiege(), new MobSpawnerTrader(iworlddataserver)); // Paper - This list is unused as we make these configurable in `Allow toggling special MobSpawners per world`. If this list changes, that patch likely needs updating.
```

Reply:
The list is "unused" but it is still passed to the WorldServer constructor. We could pass an empty list to the constructor, or remove the parameter, but either way increases the diff more than is needed imo.